### ISSUE TITLE
EnumClass code generation does not support Dart 2.17 super syntax

### ISSUE THREAD

**Issue by maxlapides** [google/built_value.dart#1147]
Currently, built_value expects `EnumClass`es to look like this:

```dart
class TestEnum extends EnumClass {
  const TestEnum._(String name) : super(name);

  static Serializer<TestEnum> get serializer => _$testEnumSerializer;

  static const TestEnum yes = _$yes;
  static const TestEnum no = _$no;
  static const TestEnum maybe = _$maybe;

  static BuiltSet<TestEnum> get values => _$values;
  static TestEnum valueOf(String name) => _$valueOf(name);
}
```

As of [Dart 2.17](https://medium.com/dartlang/dart-2-17-b216bfc80c5d#cd5e), you can rewrite that like this:

```dart
class TestEnum extends EnumClass {
  const TestEnum._(super.name);

  static Serializer<TestEnum> get serializer => _$testEnumSerializer;

  static const TestEnum yes = _$yes;
  static const TestEnum no = _$no;
  static const TestEnum maybe = _$maybe;

  static BuiltSet<TestEnum> get values => _$values;
  static TestEnum valueOf(String name) => _$valueOf(name);
}
```

But, this produces an error during code generation:

```
Please make the following changes to use EnumClass:

1. Have exactly one constructor: const TestEnum._(String name) : super(name);
```

**Comment by davidmorgan** [comment]
Thanks!

`2.17` also adds enums that support code, which should mostly make `EnumClass` unnecessary ... except that we'll still want them to be serializable. I need to take a deeper look...

**Comment by Rimantovas** [comment]
Is there a workaround?

**Comment by davidmorgan** [comment]
No, you'll have to write it the old way for now.

**Comment by kuhnroyal** [comment]
@davidmorgan Do you think you can find a quick way to allow the super syntax when you cut a release?

**Comment by davidmorgan** [comment]
Done :) will be released in `8.4.0` today. Thanks for commenting :)

**Comment by kuhnroyal** [comment]
Awesome, thanks!
